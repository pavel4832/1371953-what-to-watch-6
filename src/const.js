export const RATING_STARS = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

export const COLUMN_QUANTITY = 2;

export const PRECISION_RATING = 2;

export const TIMEOUT_PREVIEW = 1000;

export const CONTENT_TYPE = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

export const POSTER_SIZE = {
  WIDTH: `280`,
  HEIGHT: `175`
};

export const COUNT_CARD = {
  SAME: 4,
  MAIN_PER_STEP: 8,
  ACTIVE: 0
};

export const FILTER_TYPE = {
  ALL_GENRE: `All genre`,
  COMEDIES: `Comedies`,
  CRIME: `Crime`,
  DOCUMENTARY: `Documentary`,
  DRAMAS: `Dramas`,
  HORROR: `Horror`,
  KIDS_FAMILY: `kids & Family`,
  ROMANCE: `Romance`,
  SCI_FI: `Sci-Fi`,
  THRILLERS: `Thrillers`
};

export const MAX_FILTERS = 9;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  FILMS: `/films`,
  PLAYER: `/player/:id`,
};

export const APIRoute = {
  FILMS: `/films`,
  PROMO: `/films/promo`,
  COMMENTS: `/comments`,
  LOGIN: `/login`,
  ERROR: `/error`
};

export const EMAIL_REGEXP = /^(([^<>()[\].,;:\s@"]+(\.[^<>()[\].,;:\s@"]+)*)|(".+"))@(([^<>()[\].,;:\s@"]+\.)+[^<>()[\].,;:\s@"]{2,})$/iu;
