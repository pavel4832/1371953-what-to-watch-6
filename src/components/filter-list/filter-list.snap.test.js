import React from 'react';
import {Provider} from 'react-redux';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import FilterList from './filter-list';
import configureStore from 'redux-mock-store';
import MOVIES from '../../mock/movies';
import {FILTER_TYPE} from '../../const';

const mockStore = configureStore({});

it(`Should FilterList render correctly`, () => {
  const store = mockStore({
    DATA: {movies: MOVIES, genre: FILTER_TYPE.ALL_GENRE},
  });
  const history = createMemoryHistory();

  const {container} = render(
      <Provider store={store}>
        <Router history={history}>
          <FilterList />
        </Router>
      </Provider>
  );

  expect(container).toMatchSnapshot();
});
