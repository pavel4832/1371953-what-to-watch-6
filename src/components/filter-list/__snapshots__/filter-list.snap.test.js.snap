// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilterList render correctly 1`] = `
<div>
  <ul
    class="catalog__genres-list"
  >
    <li
      class="catalog__genres-item catalog__genres-item--active"
    >
      <a
        class="catalog__genres-link"
        href="#"
      >
        All genre
      </a>
    </li>
    <li
      class="catalog__genres-item "
    >
      <a
        class="catalog__genres-link"
        href="#"
      >
        Comedies
      </a>
    </li>
    <li
      class="catalog__genres-item "
    >
      <a
        class="catalog__genres-link"
        href="#"
      >
        Dramas
      </a>
    </li>
    <li
      class="catalog__genres-item "
    >
      <a
        class="catalog__genres-link"
        href="#"
      >
        Sci-Fi
      </a>
    </li>
    <li
      class="catalog__genres-item "
    >
      <a
        class="catalog__genres-link"
        href="#"
      >
        Thrillers
      </a>
    </li>
  </ul>
</div>
`;
